code/modules/clothing/under/_under.dm

```diff
@@ -121,56 +121,65 @@
-		var/obj/item/clothing/accessory/A = I
-		if(attached_accessory)
-			if(user)
-				to_chat(user, "<span class='warning'>[src] already has an accessory.</span>")
-			return
-		else
+	if(!istype(I, /obj/item/clothing/accessory))
+		return
+	var/obj/item/clothing/accessory/A = I
+	if(attached_accessory)
+		if(user)
+			to_chat(user, "<span class='warning'>[src] already has an accessory.</span>")
+		return
 
-			if(!A.can_attach_accessory(src, user)) //Make sure the suit has a place to put the accessory.
-				return
-			if(user && !user.temporarilyRemoveItemFromInventory(I))
-				return
-			if(!A.attach(src, user))
-				return
+	if(!A.can_attach_accessory(src, user)) //Make sure the suit has a place to put the accessory.
+		return
+	if(user && !user.temporarilyRemoveItemFromInventory(I))
+		return
+	if(!A.attach(src, user))
+		return
 
-			if(user && notifyAttach)
-				to_chat(user, "<span class='notice'>You attach [I] to [src].</span>")
+	. = TRUE
+	if(user && notifyAttach)
+		to_chat(user, "<span class='notice'>You attach [I] to [src].</span>")
 
-			var/accessory_color = attached_accessory.icon_state
-			accessory_overlay = mutable_appearance('icons/mob/clothing/accessories.dmi', "[accessory_color]")
-			accessory_overlay.alpha = attached_accessory.alpha
-			accessory_overlay.color = attached_accessory.color
+	var/accessory_color = attached_accessory.icon_state
+	accessory_overlay = mutable_appearance('icons/mob/clothing/accessories.dmi', "[accessory_color]")
+	accessory_overlay.alpha = attached_accessory.alpha
+	accessory_overlay.color = attached_accessory.color
 
-			if(ishuman(loc))
-				var/mob/living/carbon/human/H = loc
-				H.update_inv_w_uniform()
-				H.update_inv_wear_suit()
-				H.fan_hud_set_fandom()
+	update_appearance()
+	if(!ishuman(loc))
+		return
 
-			return TRUE
+	var/mob/living/carbon/human/H = loc
+	H.update_inv_w_uniform()
+	H.update_inv_wear_suit()
+	H.fan_hud_set_fandom()
 
 /obj/item/clothing/under/proc/remove_accessory(mob/user)
+	. = FALSE
 	if(!isliving(user))
 		return
 	if(!can_use(user))
 		return
 
-	if(attached_accessory)
-		var/obj/item/clothing/accessory/A = attached_accessory
-		attached_accessory.detach(src, user)
-		if(user.put_in_hands(A))
-			to_chat(user, "<span class='notice'>You detach [A] from [src].</span>")
-		else
-			to_chat(user, "<span class='notice'>You detach [A] from [src] and it falls on the floor.</span>")
+	if(!attached_accessory)
+		return
 
-		if(ishuman(loc))
-			var/mob/living/carbon/human/H = loc
-			H.update_inv_w_uniform()
-			H.update_inv_wear_suit()
-			H.fan_hud_set_fandom()
+	. = TRUE
+	var/obj/item/clothing/accessory/A = attached_accessory
+	attached_accessory.detach(src, user)
+	if(user.put_in_hands(A))
+		to_chat(user, "<span class='notice'>You detach [A] from [src].</span>")
+	else
+		to_chat(user, "<span class='notice'>You detach [A] from [src] and it falls on the floor.</span>")
+
+	update_appearance()
+	if(!ishuman(loc))
+		return
+
+	var/mob/living/carbon/human/H = loc
```

Comment:
https://github.com/tgstation/tgstation/blob/master/.github/CONTRIBUTING.md#use-descriptive-and-obvious-names